ext/mbstring/libmbfl/filters/mbfilter_uuencode.c

```diff
@@ -147,3 +149,17 @@ int mbfl_filt_conv_uudec(int c, mbfl_convert_filter * filter)
 	}
 	return 0;
 }
+
+static int mbfl_filt_conv_uudec_flush(mbfl_convert_filter *filter)
+{
+	if (filter->status == uudec_state_ground) {
+		/* We never found any UUencoded data in the input */
+		(*filter->output_function)(MBFL_BAD_INPUT, filter->data);
+	}
+
+	if (filter->flush_function) {
+		(*filter->flush_function)(filter->data);
+	}
+
+	return 0;
+}
```

Comment:
> OK. Sounds good.
> 
> The uuencode change is (indirectly) tested by the added tests for `mb_detect_encoding`. Not good enough?

As this also affects encoding conversion, it would be good to test explicitly. My understanding is that it will now return a replacement character for cases where it previously returned an empty string?

> If we blacklist uuencode, then maybe there's no need to add this flush function?

Yeah, probably. The uuencode filter looks pretty broken in a way that may not be salvageable, so if we can just ignore it here, we don't need to try and make it more sensible. We should add a deprecation for these non-encodings on master and then remove them entirely at some point.

Reply:
As we maintain the mbstring extension, this causes technical issues and the impact is expected to be low, I don't think that an RFC is necessary.